src/transformime/transformers.ts

```diff
@@ -66,24 +101,40 @@ class JavascriptTransformer implements ITransformer<Widget> {
   }
 }
 
+
+/**
+ * A transformer for `<svg>` data.
+ */
 export
 class SVGTransformer implements ITransformer<Widget> {
   mimetypes = ['image/svg+xml'];
   transform(mimetype: string, data: string): Widget {
     let w = new Widget();
     w.node.innerHTML = data;
+    let svgElement = w.node.getElementsByTagName('svg')[0];
+    if (!svgElement) {
+      throw new Error("SVGTransform: Error: Failed to create <svg> element");
+    }
     return w;
   }
 }
 
+
+/**
+ * A transformer for LateX data.
+ */
 export
 class LatexTransformer implements ITransformer<Widget> {
   mimetypes = ['text/latex'];
   transform(mimetype: string, data: string): Widget {
     let w = new Widget();
     w.node.textContent = data;
-    // TODO: do something to the class to get MathJax to render it?
-    // TODO: use katex if mathjax isn't defined?
+    if (typeof MathJax !== "undefined") {
+      w.node.textContent = data;
+      MathJax.Hub.Queue(["Typeset", MathJax.Hub, w.node]);
```

Comment:
Not each time.  However, it does need to be shown, IIRC (not display none). Are we guaranteed the widget is visible in onAfterAttach?